pkg/controller/elasticsearch/filesettings/reconciler.go

```diff
@@ -60,18 +61,18 @@ func ReconcileSecret(
 	ctx context.Context,
 	c k8s.Client,
 	expected corev1.Secret,
-	es esv1.Elasticsearch,
+	owner client.Object,
 ) error {
 	reconciled := &corev1.Secret{}
 	return reconciler.ReconcileResource(reconciler.Params{
 		Context:    ctx,
 		Client:     c,
-		Owner:      &es,
+		Owner:      owner,
 		Expected:   &expected,
 		Reconciled: reconciled,
 		NeedsUpdate: func() bool {
-			return !maps.IsSubset(expected.Labels, reconciled.Labels) ||
-				!maps.IsSubset(expected.Annotations, reconciled.Annotations) ||
+			return !reflect.DeepEqual(expected.Labels, reconciled.Labels) ||
```

Comment:
@pebrc 
Two things why we can't/do not need to use IsSubset here
1) These are operator created settings not user created ones, so there are unlikely to be additional labels and annotations.
2) more importantly though, this leads to a bug that is currently present even with elasticsearch filesettings, consider this scenario
say currently the secret has the following annotations
```
annotations:
    policy.k8s.elastic.co/secure-settings-secrets: '[{"namespace":"default","secretName":"oidc-secret"}]'
    policy.k8s.elastic.co/settings-hash: "4284554765"
```
And we want to remove the secure settings annotations, because the secure setting is removed from the stackconfigpolicy,  so we want the new annotations to be the following
```
annotations:
    policy.k8s.elastic.co/settings-hash: "4284554765"
```
With the check that was in place
!maps.IsSubset(expected.Annotations, reconciled.Annotations)
expected annotations is a subset of reconciled annotations, so this is going to be false and we will never update the secret. So during my testing I found that removing secureSettings from the SCP had no effect

 instead if we check for equality here and then in the update function we merge the annotations and remove only the managed annotations (which is what we do) I think it still achieves what we want with preserving additional labels and annotations introduced by third party systems.

Reply:
I agree with @barkbay and in this particular case could you not just use an empty array to reset the annotation instead of removing it? That would be compatible with `IsSubSet` as it also compares values.

Reply:
@barkbay @pebrc I understand what we are trying to do here and that we want to preserve labels and annotations not added by the operator and not modify it, but the change does not change behaviour right ? even though we check for equality, in the upgrade function we do this 
```
reconciled.Labels = maps.Merge(reconciled.Labels, expected.Labels)
			// remove managed labels if they are no longer defined
			for _, label := range managedLabels {
				if _, ok := expected.Labels[label]; !ok {
					delete(reconciled.Labels, label)
				}
			}
			reconciled.Annotations = maps.Merge(reconciled.Annotations, expected.Annotations)
			// remove managed annotations if they are no longer defined
			for _, annotation := range managedAnnotations {
				if _, ok := expected.Annotations[annotation]; !ok {
					delete(reconciled.Annotations, annotation)
				}
			}
			reconciled.Data = expected.Data
```
which should preserve the user added labels etc right ? I did not fully understand the concern.

Reply:
cleared this up offline with @pebrc and @thbkrkr , will add a new function that deals with only checking a specific set of labels

Reply:
@pebrc @barkbay @thbkrkr I add a new function that you can pass in a set of managed keys and only those will be compared. Please take a look. TIA